chain/rust/src/builders/tx_builder.rs

```diff
@@ -726,16 +803,27 @@ impl TransactionBuilder {
                     .for_each(|signer| self.add_required_signer(*signer));
 
                 match &script_witness.script {
-                    PlutusScriptWitness::Ref(ref_script) => {
+                    PlutusScriptWitness::Ref(ref_script_hash) => {
+                        // it could also be a reference script - check those too
+                        // TODO: do we want to change how we store ref scripts to cache the hash to avoid re-hashing (slow on wasm) every time an input is added?
```

Comment:
TODO? Okay to merge with this?

Reply:
With this, yes. I don't think it's too bad of a performance hit for now (did not profile it - just going off past experience with hashing in the builder and we don't support asm.js in the new CML anyway so it's not as ridiculous a difference) and it's only when using ref scripts and we can always optimize later, using this comment as a reminder.